iroh-relay/src/relay_map.rs

```diff
@@ -52,50 +51,81 @@ impl RelayMap {
-            }
-            .into(),
-        );
-
-        RelayMap {
-            nodes: Arc::new(nodes),
+impl FromIterator<RelayNode> for RelayMap {
+    fn from_iter<T: IntoIterator<Item = RelayNode>>(iter: T) -> Self {
+        Self {
+            nodes: Arc::new(
+                iter.into_iter()
+                    .map(|node| (node.url.clone(), Arc::new(node)))
+                    .collect(),
+            ),
         }
     }
+}
 
-    /// Returns a [`RelayMap`] from a [`RelayUrl`].
-    ///
-    /// This will use the default STUN port, the default QUIC port
-    /// (as defined by the `iroh-relay` crate) and IP addresses
-    /// resolved from the URL's host name via DNS.
-    /// relay nodes are specified at <../../docs/relay_nodes.md>
-    pub fn from_url(url: RelayUrl) -> Self {
-        Self::default_from_node(url, DEFAULT_STUN_PORT)
-    }
-
-    /// Constructs the [`RelayMap`] from an iterator of [`RelayNode`]s.
-    pub fn from_nodes<I: Into<Arc<RelayNode>>>(value: impl IntoIterator<Item = I>) -> Result<Self> {
-        let mut map = BTreeMap::new();
-        for node in value.into_iter() {
-            let node = node.into();
-            ensure!(!map.contains_key(&node.url), "Duplicate node url");
-            map.insert(node.url.clone(), node);
+/// Create a `RelayMap` from a `RelayUrl`.
+///
+/// The `RelayNode`s in the `RelayMap` will have the default stun and quic address
+/// discovery ports.
+impl From<RelayUrl> for RelayMap {
+    fn from(value: RelayUrl) -> Self {
+        Self {
+            nodes: Arc::new([(value.clone(), Arc::new(value.into()))].into()),
+        }
+    }
+}
+
+/// Create a `RelayMap` from a `RelayUrl` and a stun port.
+///
+/// The resulting `RelayMap` contains `RelayNode`s with the default quic address
+/// discovery port.
+impl From<(RelayUrl, u16)> for RelayMap {
+    fn from(value: (RelayUrl, u16)) -> Self {
+        Self {
+            nodes: Arc::new([(value.0.clone(), Arc::new(value.into()))].into()),
+        }
+    }
+}
+
+impl From<RelayNode> for RelayMap {
+    fn from(value: RelayNode) -> Self {
+        Self {
+            nodes: Arc::new([(value.url.clone(), Arc::new(value))].into()),
+        }
+    }
+}
+
+/// Create a `RelayMap` from an iterator of `RelayUrl`.
+///
+/// The `RelayNode`s in the `RelayMap` will have the default stun and quic address
+/// discovery ports.
+impl FromIterator<RelayUrl> for RelayMap {
+    fn from_iter<T: IntoIterator<Item = RelayUrl>>(iter: T) -> Self {
+        Self {
+            nodes: Arc::new(
+                iter.into_iter()
+                    .map(|url| (url.clone(), Arc::new(url.into())))
+                    .collect(),
+            ),
+        }
+    }
+}
+
+/// Create a `RelayMap` from an iterator of `RelayUrl`s and a stun ports.
+///
+/// The resulting `RelayMap` contains `RelayNode`s with the default quic address
+/// discovery port.
+impl FromIterator<(RelayUrl, u16)> for RelayMap {
```

Comment:
Like the above `(url, stun_port)` constructor I'd be fine with removing this if it doesn't cause too much breakage.